telemetry/server.go

```diff
@@ -66,58 +103,23 @@ func (s *Server) createTelemetryData(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	errorDetails := NewMetricErrors(s.logger)
+	errorDetails := common.NewMetricErrors(s.logger)
 
 	for _, data := range telemetryData {
-		switch data.TelemetryType {
-		case types.ProtocolStatsMetric:
-			var stats ProtocolStats
-			err := stats.process(s.DB, errorDetails, &data)
-			if err != nil {
-				continue
-			}
-		case types.ReceivedEnvelopeMetric:
-			var envelope ReceivedEnvelope
-			err := envelope.process(s.DB, errorDetails, &data)
-			if err != nil {
-				continue
-			}
-		case types.SentEnvelopeMetric:
-			var envelope SentEnvelope
-			err := envelope.process(s.DB, errorDetails, &data)
-			if err != nil {
-				continue
-			}
-		case types.ErrorSendingEnvelopeMetric:
-			var envelopeError ErrorSendingEnvelope
-			err := envelopeError.process(s.DB, errorDetails, &data)
-			if err != nil {
-				continue
-			}
-		case types.PeerCountMetric:
-			var peerCount PeerCount
-			err := peerCount.process(s.DB, errorDetails, &data)
-			if err != nil {
-				continue
-			}
-		case types.PeerConnFailureMetric:
-			var peerConnFailure PeerConnFailure
-			err := peerConnFailure.process(s.DB, errorDetails, &data)
-			if err != nil {
-				continue
-			}
-		case types.ReceivedMessagesMetric:
-			var receivedMessages ReceivedMessage
-			err := receivedMessages.process(s.DB, errorDetails, &data)
-			if err != nil {
-				continue
-			}
-		default:
-			errorDetails.Append(data.Id, fmt.Sprintf("Unknown telemetry type: %s", data.TelemetryType))
+		metric, ok := s.metrics[data.TelemetryType]
+		if !ok {
+			//errorDetails.Append(data.Id, fmt.Sprintf("Unknown telemetry type: %s", data.TelemetryType))
```

Comment:
Unless it's needed for something:
```suggestion
```

Reply:
Hmm, not sure why I commented it out - probably because I was not sure if we needed to return the error to the client or just log it and ignore